### ISSUE TITLE
As a developer/owner for the Azure SDK for C++, I want the nightly CI pipelines to create a github issue whenever it fails

### ISSUE THREAD

**Issue by vhvb1989** [Azure/azure-sdk-for-cpp#1061]
If a pipeline step fails (like cpp - core - validate_linux... ) , make the pipeline to automatically create a github issue adding the log in the description and with a tittle mentioning the pipeline and os configuration.

If the pipeline keeps failing, the issue should be updated with a new comment for each fail.

There is no need to make the pipeline to close the issue when on the next first pass. (let developer/owner to close either manually or with a PR).

Once the issue is closed, if there is a new fail (regression), a new issue should be created.

**Comment by ahsonkhan** [comment]
@weshaggard, do other language SDKs already have a solution for tracking nightly build failures?

Auto-submitting a PR that disables the test might be nice too, but that might encourage the wrong pattern or have too many false positives.

**Comment by weshaggard** [comment]
Currently no other languages have any automation that will automatically file issues or disable tests. We have devops and powerbi dashboards for looking at the nightly test runs but that still requires someone to investigate.

**Comment by weshaggard** [comment]
That said this would be an interesting feature but I suspect it might take some effort. I'm happy to chat about potential designs here if someone has some bandwidth to work on this.

**Comment by weshaggard** [comment]
cc @mitchdenny

**Comment by mitchdenny** [comment]
Auto filing bugs on build failures is something that Azure DevOps has had for a while (setup between Azure Boards and Azure Pipelines). But won't work OOB with GitHub. I would say though that when teams do enable it the build failure bugs are largely ignored. 

If you want to go this route then I think you've got two problems that need to be addressed. The first is the technology one which is just making the solution able to file bugs, but the second is a people problem and introducing a process into the team where these build bugs are tackled in a timely basis.

I think auto disabling tests is dangerous, but I have seen it done fairly well in the past. What happens is that when a flakey/failing test is detected a bug is raised for it after a period of time.

**Comment by ahsonkhan** [comment]
> but the second is a people problem and introducing a process into the team where these build bugs are tackled in a timely basis.

That is good to call out and we'll keep that in mind. Since GitHub issues are what we use for tracking work and bugs, having that as a single source makes the most sense and will feed into our existing triaging and tracking processes. What we don't want is unread/ignored (closed-source) emails.

The auto-generated disabling PRs would still need to be reviewed, but they help avoid us doing commonly occurring manual tasks repeatedly to unblock CI and master. Of course, the area owner and the person who wrote the test and product code needs to go and investigate and resolve the underlying issue, but other unrelated PRs being blocked hurts team productivity.

**Comment by mitchdenny** [comment]
Nothing to turn on, it was a feature that only worked with Azure Pipelines and Azure Boards. We would need to build a solution for GitHub and Azure Pipelines.

**Comment by vhvb1989** [comment]
based on engsys responses, this won't be implemented
